sql/src/main/java/io/crate/metadata/RowContextCollectorExpression.java

```diff
@@ -43,6 +43,20 @@ public void setNextRow(TRow row) {
         return forFunction(fun.andThen(BytesRefs::toBytesRef));
     }
 
+    public static <TRow> RowCollectExpression<TRow, Object> nullElseFunction(Function<TRow, Object> nullCheck,
```

Comment:
Cool idea, I wonder of we should make it even more convenient and have the second function take the return value of the first.

Maybe like `withNullableProperty(Function<TRow, T> getProperty, Function<T, Object> extractValue)`